### ISSUE TITLE
Error when compiling code with array expressions

### ISSUE THREAD

**Issue by seibert** [numba/numba#1264]
This code:
```python
import numpy as np
from numba import jit

@jit(nopython=True)
def distance_matrix(vectors):
    n_vectors = vectors.shape[0]
    result = np.empty((n_vectors, n_vectors), dtype=np.float64)

    for i in range(n_vectors):
        for j in range(i, n_vectors):
            result[i,j] = result[j,i] = np.sum((vectors[i] - vectors[j])**2)**0.5
    
    return result

n = 1000
x = np.random.uniform(size=n*3).reshape((n,3))

distance_matrix(x)
```

The exception backtrace is really long, but ends with:
```
LoweringError: Failed at nopython (nopython mode backend)
Internal error:
ValueError: '$81.6' is not a valid parameter name
File "<ipython-input-2-2f83f6734cde>", line 11
```

**Comment by gmarkall** [comment]
The error message isn't very good, but I think the issue is that broadcasting isn't supported - the `**2` is applied to every element of `(vectors[i] - vectors[j])`.

**Comment by jriehl** [comment]
The issue is that i and j are phi nodes in the array expression, so they lose their Python-compatible names and get IR names instead.  I need to do a renaming pass before trying to compile the array expression.

**Comment by sklam** [comment]
Fixed via #1271
